src/main/java/org/scijava/io/handle/DataHandle.java

```diff
@@ -540,68 +533,20 @@ default char readChar() throws IOException {
 
 	@Override
 	default int readInt() throws IOException {
-		final int ch0;
-		final int ch1;
-		final int ch2;
-		final int ch3;
-		if (isBigEndian()) {
-			ch0 = read();
-			ch1 = read();
-			ch2 = read();
-			ch3 = read();
-		}
-		else {
-			ch3 = read();
-			ch2 = read();
-			ch1 = read();
-			ch0 = read();
-		}
-		if ((ch0 | ch1 | ch2 | ch3) < 0) throw new EOFException();
-		return ((ch0 << 24) + (ch1 << 16) + (ch2 << 8) + (ch3 << 0));
+		final byte[] buf = new byte[4];
```

Comment:
Same: performance will tank. Need a small reusable `byte[]` as instance field. This should be OK, since these methods are already not thread-safe.